### DISCUSSION TITLE
Performance Experiment MQ Actor Count Question

### DISCUSSION THREAD

**Discussion by thesammiller** [CS-6381/FinalProject#27] [Q&A]
Using Redis, I'm able to work through the first experiment of the Actor Count. Processing the information afterwards is fairly straightforward when it's 1 producer. 

I'm not sure how to check times with 5 producers and 1 sub. It seems like we'd need to know which message came from which producer, but then we'd have to include some details of who sent the message. This is easy to implement, but it would increase the message size and need to be standardized across experiments. 

Any other suggestions for approaching this?

@jmbeach  @laynemoseley

**Comment by laynemoseley** [comment]
@thesammiller I think you'll have to send a timestamp with each message. Otherwise there will be no way to know how long it took to receive.

You could even put the timestamp as the _very first few bytes_ of the message, and then you subs could parse the message up to the first space to read out the timestamp. This will add a very minimal amount to each message.

```
1618181031 <contents of message>
```

**Comment by jmbeach** [comment]
@thesammiller. You're right. I completely overlooked this. You'll need to send the message ID as part of every message or there is no way to marry up the output between the two producer and consumer. I will update the tests to reflect this.
